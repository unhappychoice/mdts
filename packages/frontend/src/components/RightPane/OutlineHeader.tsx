import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import { Box, IconButton, Typography } from '@mui/material';
import React from 'react';

interface OutlineHeaderProps {
  isOpen: boolean;
  onToggle: () => void;
}

const OutlineHeader: React.FC<OutlineHeaderProps> = ({ isOpen, onToggle }) => {
  return (
    <Box
      sx={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'center',
        height: '34px',
        marginBottom: 2,
        px: isOpen ? 2 : 0.5,
      }}
    >
      <IconButton onClick={onToggle} size="small" sx={{ marginBottom: 0, marginLeft: isOpen ? '0' : '12px' }} aria-label="toggle outline">
        {isOpen ? <ChevronRightIcon /> : <ChevronLeftIcon />}
      </IconButton>
      {isOpen && (
        <Typography variant="h6" gutterBottom sx={{ marginLeft: 1, marginBottom: 0 }}>
          Outline
        </Typography>
      )}
    </Box>
  );
};

export default OutlineHeader;
