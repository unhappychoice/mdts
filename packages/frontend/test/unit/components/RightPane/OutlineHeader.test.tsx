import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import OutlineHeader from '../../../../src/components/RightPane/OutlineHeader';

describe('OutlineHeader', () => {
  test('renders correctly when open', () => {
    const { asFragment } = render(
      <OutlineHeader isOpen={true} onToggle={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Outline')).toBeInTheDocument();
  });

  test('renders correctly when closed', () => {
    const { asFragment } = render(
      <OutlineHeader isOpen={false} onToggle={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.queryByText('Outline')).not.toBeInTheDocument();
  });

  test('calls onToggle when toggle button is clicked', () => {
    const onToggleMock = jest.fn();
    render(
      <OutlineHeader isOpen={true} onToggle={onToggleMock} />
    );
    fireEvent.click(screen.getByTestId('ChevronRightIcon'));
    expect(onToggleMock).toHaveBeenCalled();
  });
});
