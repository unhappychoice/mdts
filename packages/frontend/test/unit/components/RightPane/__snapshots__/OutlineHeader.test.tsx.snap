// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`OutlineHeader renders correctly when closed 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1lsng3a"
  >
    <button
      aria-label="toggle outline"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-130mf42-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="ChevronLeftIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`OutlineHeader renders correctly when open 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1nqjmij"
  >
    <button
      aria-label="toggle outline"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1nyqon6-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="ChevronRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
      </svg>
    </button>
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-onjnbs-MuiTypography-root"
    >
      Outline
    </h6>
  </div>
</DocumentFragment>
`;
